### ISSUE TITLE
Unable to read python formatter.

### ISSUE THREAD

**Issue by jtw023** [ChristianChiarulli/LunarVim#341]
When I do `which yapf` in my terminal emulator I get back `/home/jordan/.local/bin/yapf` which I put into my `/home/jordan/.config/nvim/lv-settings.lua` and it causes this error every time I open a file: 
~~~
E5113: Error while calling lua chunk: /home/jordan/.config/nvim/init.lua:4: Vim(luafile):E5113: Error while ca
lling lua chunk: /home/jordan/.config/nvim/lv-settings.lua:31: attempt to index field 'yapf' (a nil value)
~~~
I've also tried removing the 'yapf' part so it ends at `/home/jordan/.local/bin/` and that gave me the same error.

RECAP: In my lv-settings.lua I had tried to just install yapf and flake8 as I had for the `lua-format` program but still python formatting and linting did not seem to be working at all. After that, I decided to try as those comments said and add the exec path which then introduced that error. 

Does anyone know the solution?

Here is my lv-settings.lua python section.
![2021-05-01_18-15](https://user-images.githubusercontent.com/59935328/116798929-480df000-aaa9-11eb-879c-e942c1ad65d0.png)

**Comment by rebuilt** [comment]
Remove the exec_path lines and try 
```bash
:LspInstall efm
:LspInstall python 
```
to reinstall the language servers.

**Comment by jtw023** [comment]
I tried that and now it's back to where I started, it simply won't format. If I have a situation like the below picture and I try hitting `space -> l(+lsp) -> f(format)` nothing happens. The linting works as expected now but the yapf formatting doesn't work. I also tried the black formatter and that doesn't work either.

![2021-05-02_10-47](https://user-images.githubusercontent.com/59935328/116822508-0bd3a180-ab34-11eb-80f7-53f80304fafb.png)

**Comment by rebuilt** [comment]
Does :LspInfo look like this?

![lspclients](https://user-images.githubusercontent.com/16025007/116823186-5f211080-ab83-11eb-95b3-fba0382eebf3.png)

**Comment by jtw023** [comment]
Okay, so the problem seems to be that only pyright is getting attached to the buffer. Efm is nowhere to be found.

![2021-05-02_11-27](https://user-images.githubusercontent.com/59935328/116823450-6d4a3f00-ab39-11eb-9bbc-9fe0c53b4d3a.png)

**Comment by rebuilt** [comment]
You're probably missing the error message when installing efm.  First install go, Google's go-lang.  Then retry :LspInstall efm

**Comment by jtw023** [comment]
I definitely wasn't getting an error. I ran through the installer and then verified it was installed with `go version` leading to the output in the first picture. Then re-ran `:LspInstall efm` and it installed without errors again only to not show up in `:LspInfo`.

I also included the output of `:LspInstall efm` in the second picture.

![2021-05-02_11-47](https://user-images.githubusercontent.com/59935328/116823922-2b6ec800-ab3c-11eb-8f9e-6c709336f4ad.png)

![2021-05-02_11-51](https://user-images.githubusercontent.com/59935328/116824023-cc5d8300-ab3c-11eb-9f06-06a448fa0290.png)

**Comment by rebuilt** [comment]
Try manually starting.  :LspStart efm.  Then :messages to see why it's failing to start

**Comment by jtw023** [comment]
After running `:LspStart efm`  I got this error `Autostart for efm failed: matching root directory not detected.`

I then copied my config git folder over to a test directory and everything works perfectly! Seems I just can't have autoformat on when not working in a git directory. It probably does the same thing for lua and anything else I format but the only lua i've got on my system is in the config git directory so I didn't notice it.
